### PR TITLE
Handle 404s in WorkspaceContainer, ignore everywhere else

### LINKED ISSUE #329: Infinite spinner when linking to a workspace with no access
Reproduction instructions:

When logged out of Saturn, follow a link to a workspace. Eg:
https://bvdp-saturn-prod.appspot.com/#workspaces/fc-product-demo/try-saturn

Log in to Saturn with a user _who does not have access to that linked workspace_

See an infinite spinner. Console displays a 404 from Rawls in response to the workspace get. An error banner is shown indicating a problem loading workspace details. However, you see the header info for the workspace, as if it did exist!

Preferred behavior:
Users cannot distinguish between:
A) Workspaces that exist to which the lack access
B) Workspaces do not exist

Users do not see a spinner. (or even an error header). Rather, they see a message stating "Could not display workspace. Either the requested workspace does not exist, or you do not have access. If you suspect you do not have access, please contact the workspace owner." (or similar)

### PR DESCRIPTION
**Pull request by MattPutnam**
Fixes #329

### CODE REVIEW

**Review by panentheos — COMMENTED** [review]
It seems a shame to require the individual pages to have special logic to deal with this. What about introducing a wrapper component that postpones rendering until it loads the workspace details and verifies them? You could do it with a HOC that wraps the page components.

**Comment by MattPutnam** [comment]
I tried that briefly, but I didn't get it right. I'll try again.

**Comment by MattPutnam** [comment]
@panentheos @zarsky-broad @dmohs This is ready for re-review

File: src/pages/workspaces/workspace/Data.js [hunk 1]
```diff
@@ -88,7 +92,9 @@ class WorkspaceData extends Component {
       const entityMetadata = await Rawls.workspace(namespace, name).entityMetadata()
       this.setState({ entityMetadata })
     } catch (error) {
-      reportError('Error loading workspace entity data', error)
+      if (error.status !== 404) {
+        reportError('Error loading workspace entity data', error)
+      } // Ignore 404s; container handles this error
```
Comment (panentheos) [inline src/pages/workspaces/workspace/Data.js]: We shouldn't have to do this anymore, right?

File: src/pages/workspaces/workspace/WorkspaceContainer.js [hunk 2]
```diff
@@ -215,3 +208,77 @@ export default class WorkspaceContainer extends Component {
...
+          ])
+      ])
+    }
+
+    componentDidMount() {
+      this.refresh()
+    }
+
+    componentDidUpdate() {
+      StateHistory.set(this.state)
```
Comment (panentheos) [inline src/pages/workspaces/workspace/WorkspaceContainer.js]: Do we really want to store history here? This just gets overridden by the page contents anyway, and we actually do want to check that the workspace is valid.

File: src/pages/workspaces/workspace/WorkspaceContainer.js [hunk 3]
```diff
@@ -215,3 +208,77 @@ export default class WorkspaceContainer extends Component {
...
+
+      return h(WorkspaceContainer, {
+        namespace, name, activeTab, workspace,
+        title: _.isFunction(title) ? title(this.props) : title,
+        breadcrumbs: breadcrumbs(this.props),
+        refresh: async () => {
+          await this.refresh()
+          const child = this.child.current
+          const childRefresh = Object.getPrototypeOf(child).refresh
+          childRefresh && childRefresh.call(child)
```
Comment (panentheos) [inline src/pages/workspaces/workspace/WorkspaceContainer.js]: Could this just be `this.child.current.refresh()`? (with appropriate null guards)

**Review by panentheos — APPROVED** [review]
Looks good.

**Pull request merged**